lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/model/MethodDefinition.java

```diff
@@ -197,17 +203,32 @@ public String getConstructorExpression() {
      * @param position the position (0-indexed) of this argument in the arguments list.
      */
     public ArgumentDefinition(VariableElement element, int position) {
+      this.element = element;
       type = element.asType();
       String[] typeNameSegments = type.toString().split("\\.");
       typeName = typeNameSegments[typeNameSegments.length - 1];
       String originalName = element.getSimpleName().toString();
       name = originalName.equals("_this") ? "this" : originalName;
       isState = element.getAnnotation(MonadicState.class) != null && type.toString().equals(OBJECT);
+      isSuspended = element.getAnnotation(Suspend.class) != null;
       isFrame = type.toString().equals(VIRTUAL_FRAME);
       isCallerInfo = type.toString().equals(CALLER_INFO);
       this.position = position;
     }
 
+    public boolean validate(ProcessingEnvironment processingEnvironment) {
+      if (type.toString().equals(THUNK)) {
+        processingEnvironment
+            .getMessager()
+            .printMessage(
+                Diagnostic.Kind.ERROR,
+                "Argument must not be typed as Thunk. Use @Suspend instead.",
```

Comment:
This error is a touch unclear. From above it looks like you have to go `@Suspend Object foo`, not just `@Suspend foo`.